File: ReSharper.FSharp/src/Services.FSharp/src/Comment/FSharpLineCommentActionProvider.cs
using JetBrains.ReSharper.Feature.Services.Comment;
using JetBrains.ReSharper.Psi;
using JetBrains.ReSharper.Psi.FSharp;
using JetBrains.ReSharper.Psi.FSharp.Parsing;
using JetBrains.ReSharper.Psi.Parsing;

namespace JetBrains.ReSharper.Feature.Services.FSharp.Comment
{
  [Language(typeof(FSharpLanguage))]
  public class FSharpLineCommentActionProvider : SimpleLineCommentActionProvider
  {
    public override string StartLineCommentMarker => "//";

    protected override bool IsNewLine(TokenNodeType tokenType)
    {
      return tokenType == FSharpTokenType.NEW_LINE;
    }

    protected override bool IsEndOfLineComment(TokenNodeType tokenType, string tokenText)
    {
      if (tokenType != FSharpTokenType.COMMENT)
        return false;

      // exactly 2 slashes
      if (tokenText.Length == 2 || tokenText[2] != '/')
        return true;

      // comment with 4 or more slashes
      if (tokenText.Length > 3 && tokenText[3] == '/')
        return true;

      // doc comment
      return false;
    }

    protected override bool IsWhitespace(TokenNodeType tokenType)
    {
      return tokenType == FSharpTokenType.WHITESPACE;
    }
  }
}

File: ReSharper.FSharp/src/FSharp/FSharp.Psi.Features/src/Comment/FSharpLineCommentActionProvider.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Services.Comment

open JetBrains.Application.Settings
open JetBrains.ReSharper.Feature.Services.Comment
open JetBrains.ReSharper.Plugins.FSharp.Psi
open JetBrains.ReSharper.Plugins.FSharp.Psi.Parsing
open JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
open JetBrains.ReSharper.Psi

[<Language(typeof<FSharpLanguage>)>]
type FSharpLineCommentActionProvider() =
    inherit SimpleLineCommentActionProvider()

    override x.StartLineCommentMarker = "//"
    override x.IsNewLine(tokenType) = tokenType == FSharpTokenType.NEW_LINE
    override x.IsWhitespace(tokenType) = tokenType == FSharpTokenType.WHITESPACE

    override x.IsEndOfLineComment(tokenType, tokenText) =
        tokenType == FSharpTokenType.LINE_COMMENT &&
        (tokenText.Length = 2 || tokenText[2] <> '/' || tokenText.Length > 3 && tokenText[3] = '/')

    override x.ShouldInsertAtLineStart(settingsStore) =
        settingsStore.GetValue(fun (key: FSharpFormatSettingsKey) -> key.PlaceCommentsAtFirstColumn) &&
        settingsStore.GetValue(fun (key: FSharpFormatSettingsKey) -> key.StickComment)
